### PR TITLE
move codewhisperer extensionPoint to codewhisperer plugin.xml

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
## Root cause
After split, amazonq: plugin.xml not longer consume `ext-java.xml`, `ext-python.xml` ... etc. Thus extension declared within `ext-xx.xml` are not accessible from the Q plugin

## Impact
- ImportAdder not working
- UTG will throw error if the strategy is 'fetchByContent', i.e. there is not matching source file (`JavaClassTest.java` -> `JavaClass.java`)

```
ava.lang.ClassCastException: class software.aws.toolkits.jetbrains.services.codewhisperer.language.classresolver.CodeWhispereJavaClassResolver cannot be cast to class software.aws.toolkits.jetbrains.services.codewhisperer.language.classresolver.CodeWhispererClassResolver (software.aws.toolkits.jetbrains.services.codewhisperer.language.classresolver.CodeWhispereJavaClassResolver is in unnamed module of loader com.intellij.ide.plugins.cl.PluginClassLoader @64ed4cea; software.aws.toolkits.jetbrains.services.codewhisperer.language.classresolver.CodeWhispererClassResolver is in unnamed module of loader com.intellij.ide.plugins.cl.PluginClassLoader @7ec66510)
    at software.aws.toolkits.jetbrains.services.codewhisperer.util.JavaCodeWhispererFileCrawler$findSourceFileByContent$1$1.invoke(JavaCodeWhispererFileCrawler.kt:89)
    at software.aws.toolkits.jetbrains.services.codewhisperer.util.JavaCodeWhispererFileCrawler$findSourceFileByContent$1.invoke$lambda$0(JavaCodeWhispererFileCrawler.kt:89)
    at com.intellij.openapi.extensions.impl.ExtensionProcessingHelper.findFirstSafe$lambda$0(ExtensionProcessingHelper.kt:49)
    at com.intellij.openapi.extensions.impl.ExtensionProcessingHelper.computeSafeIfAny(ExtensionProcessingHelper.kt:59)
    at com.intellij.openapi.extensions.impl.ExtensionProcessingHelper.findFirstSafe(ExtensionProcessingHelper.kt:49)
```

## UTG

https://github.com/aws/aws-toolkit-jetbrains/assets/96078566/cb174cf0-efa9-401b-8af5-8ba858b0b87a

## ImportAdder

https://github.com/aws/aws-toolkit-jetbrains/assets/96078566/292c6bea-24b1-4e8d-a46e-edcec47d0e97

## Scala

https://github.com/aws/aws-toolkit-jetbrains/assets/96078566/015c9bbb-f21b-4479-9c56-907942f5a82f

## Kotlin

https://github.com/aws/aws-toolkit-jetbrains/assets/96078566/fbbebb6a-4fb8-4314-9d31-5f98c9d0a221

## Rust

https://github.com/aws/aws-toolkit-jetbrains/assets/96078566/d59bd0d2-054a-4c01-ab91-67f27b62b5f5

## Shell

https://github.com/aws/aws-toolkit-jetbrains/assets/96078566/d2b004f9-6d91-46ab-8dbb-7580d482d887

## Java

https://github.com/aws/aws-toolkit-jetbrains/assets/96078566/366e61e4-4f41-40f2-a627-172e6d695815

## Python 

https://github.com/aws/aws-toolkit-jetbrains/assets/96078566/b817891b-cb1b-4b3a-a174-42c2155fb344

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by Will-ShaoHua** [comment]
AI: we likely don't need `CodeWhispererProgrammingLanguage` to be an ExtensionPoint anymore, considering remove it because we intend to simply use file extension to determine the language type instead of relying on platform language modules.

**Review by andrewyuq — APPROVED** [review]

File: plugins/amazonq/codewhisperer/jetbrains-community/resources/META-INF/plugin-codewhisperer.xml [hunk 1]
```diff
@@ -24,6 +26,14 @@
         <extensionPoint qualifiedName="amazon.q.codewhisperer.importAdder" interface="software.aws.toolkits.jetbrains.services.codewhisperer.importadder.CodeWhispererImportAdder" dynamic="true"/>
     </extensionPoints>
 
+    <depends optional="true" config-file="ext-codewhisperer-java.xml">com.intellij.modules.java</depends>
```
Comment (manodnyab) [inline plugins/amazonq/codewhisperer/jetbrains-community/resources/META-INF/plugin-codewhisperer.xml:29]: Are these files used in other services like codescan/transform?

**Comment by Will-ShaoHua** [comment]
linux 232 test report (both linux & windows failed)
https://d1ihu6zq92vp9p.cloudfront.net/2a5eda34-b326-4e20-9088-5413ba2ea09b/report.html

**Comment by Will-ShaoHua** [comment]
```

software.aws.toolkits.jetbrains.services.codewhisperer.JavaCodeWhispererFileCrawlerTest

  Test listUtgCandidate by content FAILED (5.6s)

  org.opentest4j.AssertionFailedError: 
  expected: temp:///src/Main.java
   but was: temp:///src/Class3.java
```

**Pull request merged**